packages/cli-platform-android/src/commands/buildAndroid/index.ts

```diff
@@ -0,0 +1,151 @@
+      'Both "tasks" and "mode" parameters were passed to "build" command. Using "tasks" for building the app.',
+    );
+  }
+
+  const tasks = args.tasks || ['assemble' + toPascalCase(mode)];
+
+  let gradleArgs = getTaskNames(androidProject.appName, tasks);
+
+  if (args.extraParams) {
+    gradleArgs = [...gradleArgs, ...args.extraParams];
+  }
+
+  if (args.activeArchOnly) {
+    const adbPath = getAdbPath();
+    const devices = adb.getDevices(adbPath);
+    const architectures = devices
+      .map((device) => {
+        return adb.getCPU(adbPath, device);
+      })
+      .filter(
+        (arch, index, array) => arch != null && array.indexOf(arch) === index,
+      );
+    if (architectures.length > 0) {
+      logger.info(`Detected architectures ${architectures.join(', ')}`);
+      // `reactNativeDebugArchitectures` was renamed to `reactNativeArchitectures` in 0.68.
+      // Can be removed when 0.67 no longer needs to be supported.
+      gradleArgs.push(
+        '-PreactNativeDebugArchitectures=' + architectures.join(','),
+      );
+      gradleArgs.push('-PreactNativeArchitectures=' + architectures.join(','));
+    }
+  }
+  await runPackager(args, config);
+  return build(gradleArgs, androidProject.sourceDir);
+}
+
+export function build(gradleArgs: string[], sourceDir: string) {
+  process.chdir(sourceDir);
+  const cmd = process.platform.startsWith('win') ? 'gradlew.bat' : './gradlew';
+  logger.info('Building the app...');
+  logger.debug(`Running command "${cmd} ${gradleArgs.join(' ')}"`);
+  try {
+    execa.sync(cmd, gradleArgs, {
+      stdio: 'inherit',
+      cwd: sourceDir,
+    });
+  } catch (error) {
+    throw new CLIError('Failed to build the app.', error);
+  }
+}
+
+export default {
+  name: 'build-android',
+  description: 'builds your app',
+  func: buildAndroid,
+  options: [
+    {
+      name: '--mode <string>',
+      description: "Specify your app's build variant",
+      default: 'debug',
+    },
+    {
+      name: '--no-packager',
+      description: 'Do not launch packager while building',
+    },
+    {
+      name: '--port <number>',
+      default: process.env.RCT_METRO_PORT || 8081,
+      parse: Number,
+    },
+    {
+      name: '--terminal <string>',
+      description:
+        'Launches the Metro Bundler in a new window using the specified terminal path.',
+      default: getDefaultUserTerminal(),
+    },
+    {
+      name: '--tasks <list>',
+      description:
+        'Run custom Gradle tasks. By default it\'s "assembleDebug". Will override passed mode and variant arguments.',
+      parse: (val: string) => val.split(','),
+    },
+    {
+      name: '--active-arch-only',
+      description:
+        'Build native libraries only for the current device architecture for debug builds.',
+      default: false,
+    },
+    {
+      name: '--extra-params <string>',
+      description: 'Custom properties passed to gradle build command',
+      parse: (val: string) => val.split(' '),
+    },
```

Comment:
most of these params are shared by build- and run-android. Let's make sure the commands are reusing them by extracting and importing where necessary :)